beancount-parser/examples/dump_parse.rs

```diff
@@ -1,9 +1,17 @@
 use beancount_parser::parse;
 
-fn main() {
-    let filename = std::env::args().nth(1).expect("filename argument");
-    let unparsed_file = std::fs::read_to_string(filename).expect("cannot read file");
+fn run() -> Result<(), Box<dyn std::error::Error>> {
+    let filename = std::env::args().nth(1).ok_or("filename argument")?;
+    let unparsed_file = std::fs::read_to_string(filename)?;
 
-    let ledger = parse(&unparsed_file);
+    let ledger = parse(&unparsed_file)?;
     dbg!(ledger);
+    Ok(())
+}
+
+fn main() {
+    match run() {
+        Err(e) => println!("Error: {}", e),
+        _ => {}
+    }
```

Comment:
Rust `main` functions can return results, so the logic in `run` can still be in `main`.

https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html

Reply:
The reason I have done it this way is that I want the `Display` version printed, not the `Debug` version you get when raising an error in main.

Reply:
Since it's an example, it might be an idea to use `anyhow::Result` here.